source/MaterialXCore/Value.cpp

```diff
@@ -333,6 +333,34 @@ string AggregateValue::getValueString() const
     return result;
 }
 
+bool AggregateValue::isEqual(ConstValuePtr other) const
+{
+  if (!other->isA<AggregateValue>())
```

Comment:
This change looks good to me from a high level, and the one issue that I notice is the use of 2-space indentation instead of the 4-space indentation that's standard in our C++ codebase.